src/System.Reflection/tests/FieldInfo/FieldInfo_ConstantTests.cs

```diff
@@ -11,67 +11,36 @@ namespace System.Reflection.Tests
-        [Fact]
-        public void TestSetValue_constStrField()
-        {
-            TestSetValue_constantField("constStrField", (object)"new value");
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constCharField()
+        public static IEnumerable<object[]> FieldInfoConstants_TestData()
         {
-            TestSetValue_constantField("charField", (object)'A');
+            yield return new object[] { "constIntField", 222 };
+            yield return new object[] { "constStrField", "new value" };
+            yield return new object[] { "charField", 'A' };
+            yield return new object[] { "boolField", false };
+            yield return new object[] { "floatField", 4.56 };
+            yield return new object[] { "doubleField", double.MaxValue };
+            yield return new object[] { "int64Field", long.MaxValue };
+            yield return new object[] { "byteField", byte.MaxValue };
         }
 
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constboolField()
-        {
-            TestSetValue_constantField("boolField", (object)false);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constfloatField()
-        {
-            TestSetValue_constantField("floatField", (object)4.56);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constdoubleField()
+        // Verify SetValue method cannot set value for constant Fields using Reflection
+        [Theory]
+        [MemberData(nameof(FieldInfoConstants_TestData))]
+        public void TestSetValue_ConstantField(string field, object newValue)
         {
-            TestSetValue_constantField("doubleField", (object)Double.MaxValue);
-        }
-
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constInt64Field()
-        {
-            TestSetValue_constantField("int64Field", (object)Int64.MaxValue);
-        }
+            FieldInfo fi = GetField(field);
+            FieldInfoConstantTests myInstance = new FieldInfoConstantTests();
 
-        //Verify SetValue method does not set value for constant Fields
-        [Fact]
-        public void TestSetValue_constbyteField()
-        {
-            TestSetValue_constantField("byteField", (object)byte.MaxValue);
+            Assert.NotNull(fi);
+            Assert.Equal(field, fi.Name);
+            Assert.Throws<FieldAccessException>(() => fi.SetValue(myInstance, newValue));
         }
 
-        //Verify SetValue method _does_ set value for RO Field
+        // Verify SetValue method _does_ set value for RO Field
         [Fact]
-        public void TestSetValue_roIntField()
+        public void TestSetValue_RoIntField()
```

Comment:
Nit: it'd be good to spell out the acronym used here, as at first glance it's not clear what RO means, e.g. ```SetValue_ReadOnlyIntField_Succeeds```